packages/next/next-server/server/api-utils.ts

```diff
@@ -197,50 +198,75 @@ export function sendStatusCode(res: NextApiResponse, statusCode: number) {
   return res
 }
 
+/**
+ * Try to match ETag according to RFC7232
+ * @implements https://tools.ietf.org/html/rfc7232#section-3.2
+ * @param req request object
+ * @param res response object
+ * @param body of response
+ */
+function matchETag(
```

Comment:
We have the `fresh` package installed which should handle all of this.

Reply:
See `packages/next/next-server/server/send-html.ts`.

Reply:
Fresh doesn't deal with all of this, but it replaces the `etag === req.headers['if-none-match']` part. See 5f9f336

Reply:
Is that what you had in mind?

Reply:
Yes, this LGTM.